app/controllers/works_controller.rb

```diff
@@ -0,0 +1,89 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy, :upvote]
+
+  def index
+    @movies = Work.category_list("movie")
+    @books = Work.category_list("book")
+    @albums = Work.category_list("album")
+  end
+
+  def show
+    if @work.nil?
+      render :not_found, status: :not_found
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = "Successfully created #{@work.category} #{@work.id}"
+      redirect_to works_path
+    else
+      flash.now[:warning] = "A problem occurred: Could not create #{@work.category}"
+
+      @work.errors.messages.each do |field, messages|
+        flash.now[field] = messages
+      end
+
+      render :new
+    end
+  end
+
+  def edit
+
+  end
+
+  def update
+    if @work.update(work_params)
+      flash[:success] = "Successfully updated #{@work.category} #{@work.id}"
+      redirect_to work_path(@work.id)
+    else
+      flash.now[:warning] = "A problem occurred: Could not update #{@work.category}"
+
+      @work.errors.messages.each do |field, messages|
+        flash.now[field] = messages
+      end
+      render :edit
+    end
+  end
+
+  def destroy
+    @work.votes.each do |vote|
+      vote.destroy
+    end
```

Comment:
Nice workaround for the dependency between works and votes. You might want to check out the [dependent](http://guides.rubyonrails.org/association_basics.html#dependent) argument to `has_many`.